src/app/request-balance/request-balance.service.ts

```diff
@@ -0,0 +1,80 @@
+import {Injectable} from '@angular/core'
+import {environment} from '../../environments/environment'
+import {BackendHttpClient} from '../shared/services/backend/backend-http-client.service'
+import {Observable} from 'rxjs'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RequestBalanceService {
+  path = `${environment.backendURL}/api/blockchain-api`
+
+  constructor(private http: BackendHttpClient) {
+  }
+
+  getRequest(id: string): Observable<RequestBalance> {
+    return this.http.get<RequestBalance>(`${this.path}/balance/${id}`, {}, true)
+  }
+
+  createRequest(data: CreateRequestBalanceData): Observable<RequestBalance> {
+    return this.http.post<RequestBalance>(`${this.path}/balance`, data, true)
+  }
+
+  updateRequest(id: string, data: UpdateRequestBalanceData): Observable<RequestBalance> {
+    return this.http.put<RequestBalance>(`${this.path}/balance/${id}`, data, true)
+  }
+}
+
+interface CreateRequestBalanceData {
+  client_id?: string;
+  block_number?: string;
+  chain_id?: number;
+  token_address?: string;
+  wallet_address?: string;
+  redirect_url?: string;
+  arbitrary_data?: ArbitraryData;
+  screen_config?: ScreenConfig;
+}
+
+interface UpdateRequestBalanceData {
+  wallet_address: string,
+  signed_message: string,
+}
+
+export interface RequestBalance {
+  id: string;
+  status: SendRequestStatus;
+  chain_id: number;
+  redirect_url: string;
+  token_address: string;
+  block_number: string;
```

Comment:
`BigNumber`?

Reply:
I generate these interfaces with the tool called `http://json2ts.com/` which converts backend API response directly to typescript so I can use it. Its underlying type is a string, which is correct. I can convert it to BigNumber in case I need it, but in most cases, the conversion is just an overhead. When I construct my own types of business logic, I use it as a BigNumber.